### PR TITLE
Feature/post report to bitbucket

### PR DESCRIPTION
**Pull request by littleMariMari**
Added **@zooshdigital/bitbucket-v8-coverage-report** and **@zooshdigital/bitbucket-code-insights** libraries, designed to streamline the process of uploading reports to Bitbucket pipelines.

**@zooshdigital/bitbucket-code-insights**

- Upload report to the Bitbucket API
- Uses node-fetch version 2.x.x, which is still compatible with CommonJS modules.
- Provides an interface for coverage report request bodies.

**@zooshdigital/bitbucket-v8-coverage-report**

- Uses the @zooshdigital/bitbucket-code-insights package to upload reports
- Creates the structure of the report from provided file
- Command-line utility: Exposes an executable called `bitbucket-v8-coverage-report`.
- Uses command-line-args npm package to handle arguments.

**License:** GNU General Public License v3.0.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    "name": "zoosh-bitbucket-tools",
+    "version": "1.0.0",
+    "private": true,
+    "packageManager": "yarn@3.6.3",
+    "description": "zoosh-boilerplate to upload a report to Bitbucket",
+    "author": {
+        "name": "Zoosh Group"
```
Comment (tolnai) [inline package.json]: zoosh digital

File: packages/bitbucket-code-insights/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+```
+
+Where data is an array of objects, each containing the title, type, and value fields.
+
+## Example
+
+```typescript
+import uploadReportToBitbucket from '@zooshdigital/bitbucket-code-insights';
+
+const externalId = 'your_external_id';
```
Comment (sfodor) [inline packages/bitbucket-code-insights/README.md:56]: What is this? Would it be possible to compute this from some Bitbucket env var (https://support.atlassian.com/bitbucket-cloud/docs/variables-and-secrets/) ?

File: packages/bitbucket-code-insights/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+const body = {
+  // Report body
+};
+
+uploadReportToBitbucket(externalId, body)
+  .then(() => {
+    console.log('Report uploaded successfully.');
```
Comment (sfodor) [inline packages/bitbucket-code-insights/README.md]: Can we add (prefix) "Bitbucket Code Insight" to all console logs?

File: packages/bitbucket-v8-coverage-report/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Zoosh bitbucket-v8-coverage-report
```
Comment (Szbuli) [inline packages/bitbucket-v8-coverage-report/README.md:1]: Can we include that the BITBUCKET_COMMIT and BITBUCKET_REPO_FULL_NAME env variables from https://support.atlassian.com/bitbucket-cloud/docs/variables-and-secrets/ are required, if you use docker in the pipelines you have to pass those to the image

File: packages/bitbucket-code-insights/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import fetch from 'node-fetch';
+import { HttpProxyAgent } from 'http-proxy-agent';
+import { BitbucketReportBody } from "./types"
+
+async function uploadReportToBitbucket(externalId: string, body: BitbucketReportBody){
+
+  const url = `http://api.bitbucket.org/2.0/repositories/${process.env.BITBUCKET_REPO_FULL_NAME}/commit/${process.env.BITBUCKET_COMMIT}/reports/${externalId}`;
+
+  console.log('url', url);
```
Comment (sfodor) [inline packages/bitbucket-code-insights/src/index.ts]: Can we add (prefix) "Bitbucket Code Insight" to all console logs?

File: packages/bitbucket-code-insights/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  const response = await fetch(url, {
+    agent: new HttpProxyAgent('http://localhost:29418'),
```
Comment (sfodor) [inline packages/bitbucket-code-insights/src/index.ts]: What is this port? Can we have a const for it? Is something we want to hard code?

File: packages/bitbucket-v8-coverage-report/index.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+          title: 'Branches',
+          type: 'PERCENTAGE',
+          value: coverageResults.total.branches.pct,
+        },
+      ],
+    };
+
+    await uploadReportToBitbucket(name, body);
+  } catch (error) {
+    console.error('Error uploading report:', error);
```
Comment (sfodor) [inline packages/bitbucket-v8-coverage-report/index.ts]: Can we add (prefix) "Bitbucket v8 Coverage Report" to all console logs?

File: packages/bitbucket-v8-coverage-report/index.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    const { name, path: reportPath } = args;
+
+    if (!name || !reportPath) {
+      throw new Error('Bitbucket v8 Coverage Report - Usage: uploadReport -n <report-name> -p <report-path>');
+    }
+
+    const coverageResults = JSON.parse(await fs.readFile(reportPath, 'utf8'));
+
+    const body = {
+      title: 'Coverage',
```
Comment (Szbuli) [inline packages/bitbucket-v8-coverage-report/index.ts]: Title should be the name as well

File: packages/bitbucket-v8-coverage-report/index.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+      report_type: 'COVERAGE',
+      details: 'Coverage report for the build.',
```
Comment (Szbuli) [inline packages/bitbucket-v8-coverage-report/index.ts]: Add name instead of build

File: packages/bitbucket-code-insights/README.md (hunk 3 above)
Comment (Szbuli) [reply packages/bitbucket-code-insights/README.md]: I think we should also add the externalId/name to the logs

File: package.json [hunk 10]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "zoosh-bitbucket-tools",
+    "version": "1.0.0",
+    "private": true,
+    "packageManager": "yarn@3.6.3",
+    "description": "zoosh-boilerplate to upload a report to Bitbucket",
```
Comment (sfodor) [inline package.json]: Do we need this? Is this related to boiler plate?

File: packages/bitbucket-v8-coverage-report/README.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+Zoosh `bitbucket-v8-coverage-report` is a boilerplate package designed to facilitate the upload of reports to Bitbucket. This package provides functions to automate the process of uploading reports generated during build processes.
```
Comment (sfodor) [inline packages/bitbucket-v8-coverage-report/README.md]: Is this a boilerplate package?

**Review by Szbuli — APPROVED** [review]
lgtm
